src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriver.java

```diff
@@ -3,16 +3,20 @@
 import edu.kis.legacy.drawer.panel.DrawPanelController;
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
+import edu.kis.powp.appbase.Application;
 import edu.kis.powp.jobs2d.Job2dDriver;
 
+
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class LineDriver extends DrawPanelController implements Job2dDriver {
 	private int startX = 0, startY = 0;
 
-	public MyAdapter() {
+	public LineDriver(Application application) {
 		super();
+
+		this.initialize(application.getFreePanel());
```

Comment:
Coupling with Application class is unnecessary. The correct solution should issue drawLine calls to a different object.